### ISSUE TITLE
sort-keys - include/exclude objects by name

### ISSUE THREAD

**Issue by goldylucks** [eslint/eslint#11823]
**What rule do you want to change?**
sort-keys

**Does this change cause the rule to produce more or fewer warnings?**
fewer potentially

**How will the change be implemented? (New option, new default behavior, etc.)?**
new option

**Please provide some example code that this change will affect:**
```js
/*eslint sort-keys: ["error", { includeObjects: ["includeMe"] }]*/

// Examples of incorrect code for this rule:
const includeMe = { b: 'b', a: 'a' }

// Exampes of correct code for this rule
const someObj = { b: 'b', a: 'a' }

/*eslint sort-keys: ["error", { excludeObjects: ["excludeMe"] }]*/

// Examples of incorrect code for this rule:
const foo = { b: 'b', a: 'a' }

// Exampes of correct code for this rule
const excludeMe = { b: 'b', a: 'a' }

```

**What does the rule currently do for this code?**
Answered above

**What will the rule do after it's changed?**
Answered above

**Are you willing to submit a pull request to implement this change?**
I will try with some guidance, I've never looked into eslint code.

**Motivation**
The biggest use case for me is sorting JSS properties, while NOT linting the rest of the app.

**Comment by platinumazure** [comment]
Hi @goldylucks, thanks for the issue.

Have you tried using [ESLint disable comments](https://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments)? I think that would be a pretty good fit for your use case, especially in the exclude case.

**Comment by goldylucks** [comment]
hi @platinumazure, thanks for the quick response man!

Disabling through comments won't work, since I'll have to disable each and every file, and only enable the styles part (what I referred tas `includeMe` above)

Any other ideas?

**Comment by goldylucks** [comment]
that's a shame, seems like quite an easy thing to implmenet, and it's there in stylelint :/

**Comment by kaicataldo** [comment]
To give some more context around my 👎🏼, I don’t think using an arbitrary name when declaring a variable is a great way to enforce a rule like this because of how brittle it would be.

Marking an object with a comment is more explicit, isn’t reliant on tribal knowledge, and can’t be broken by a variable name change.

**Comment by platinumazure** [comment]
@goldylucks You could write a custom rule for this case if you wanted. You could even use [eslint-rule-composer](https://github.com/not-an-aardvark/eslint-rule-composer) which would allow you to pull in the core rule, get its error messages, and then filter the error messages out if they correspond to an object variable name that you want to have the rule ignore.

I agree with @kaicataldo's reasoning, when it comes to why I think we shouldn't add this rule in core. It could encourage bad patterns if people aren't careful about it, and disable comments usually make more sense for most users. But, if that doesn't work for you, you always have the right to make your own rule and have ESLint behave how you want it to.

**Comment by goldylucks** [comment]
so I ended up including instead of excluding. Not very clean, but it works :)
